VeganVirus/VeganVirus/SoundAction.h

```diff
@@ -8,22 +8,23 @@
 #include <ctime>  
 
 #define MUSIC_ICON_PATH L"music.ico"
-#define NUM_OF_SONGS 1
-
+#define NUM_OF_SONGS 6
+#define SOUND_DUARTION 120
 
 
 class SoundAction : public Action
 {
 public:
 	SoundAction(double req);
-	void start();
+	virtual void start();
+	virtual void update(double dt);
 
 	bool canActivate();
 private:
 	AudioManager _audioManager;
-	clock_t songStartTime;
+	double _timer;
 	int currentSong = -1;
```

Comment:
Missing underscore. Also, I suggest adding a `#define` to the value of -1, like I wrote in another comment.

```suggestion
	int _currentSong = NO_SONG;
```